### PR TITLE
Fix link inconsistencies in man pages

### PR DESCRIPTION
**Pull request by edsantiago**
Found via:

    echo for i in docs/*.md;do x=$(perl -ne 'if (/\[(podman-.*?)\(1\)\]\((podman-.*?)\.1\.md/) { print "  $1 != $2\n" if $1 ne $2; print " ENOENT $2\n" unless -e "./docs/$2.1.md"}' <$i); if [ -n "$x" ]; then echo $i; echo "$x";fi;done

...which is probably a good candidate for another CI hook,
except I have no idea how to rewrite it in awk.

There's one remaining inconsistency: podman-refresh in podman.1.

Signed-off-by: Ed Santiago <santiago@redhat.com>

### CODE REVIEW

**Comment by mheon** [comment]
We need to deprecate refresh. It used to have a purpose, but no longer...

**Comment by edsantiago** [comment]
@mheon should I preemptively just remove it from `podman.1.md` as part of this PR, then leave it as TBD for someone else to remove from the command list and `podman-container.1.md` ?

**Comment by mheon** [comment]
@edsantiago SGTM

**Comment by edsantiago** [comment]
So done, thanks. This breaks the in-development `hack/podman-commands.sh` script, but that's probably a good thing.

There's another problem with my PR: removing `podman-help` from `podman.1.md` also breaks the compare script. It also doesn't show "podman help" as a command in the man page, but since it does document `--help`, and since it truly offends me to link to a nonexistent `podman-help(1)`, I think the PR in its current state is the least ugly solution. But as always, there may be something I've missed.

File: docs/podman-system.1.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ The system command allows you to manage the podman systems
 
 | Command  | Man Page                                            | Description                                                                  |
 | -------  | --------------------------------------------------- | ---------------------------------------------------------------------------- |
-| info     | [podman-system-info(1)](podman-info.1.md)           | Displays Podman related system information.                                  |
+| info     | [podman-info(1)](podman-info.1.md)                  | Displays Podman related system information.                                  |
```
Comment (rhatdan) [inline docs/podman-system.1.md]: This is wrong there is a podman-system-info.

File: docs/podman.1.md [hunk 2]
```diff
@@ -137,7 +137,7 @@ the exit codes follow the `chroot` standard, see below:
 | [podman-exec(1)](podman-exec.1.md)        | Execute a command in a running container.                                      |
 | [podman-export(1)](podman-export.1.md)    | Export a container's filesystem contents as a tar archive.                     |
 | [podman-generate(1)](podman-generate.1.md)| Generate structured data based for a containers and pods.                      |
-| [podman-help(1)](podman-history.1.md)     | Show help information on podman.                                               |
+| [podman(1)](podman.1.md)                  | Show help information on podman.                                               |
```
Comment (rhatdan) [inline docs/podman.1.md]: Alphabetize

File: docs/podman-container.1.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ The container command allows you to manage containers
 | create     | [podman-create(1)](podman-create.1.md)              | Create a new container.                                                      |
 | diff       | [podman-diff(1)](podman-diff.1.md)                  | Inspect changes on a container or image's filesystem.                        |
 | exec       | [podman-exec(1)](podman-exec.1.md)                  | Execute a command in a running container.                                    |
-| exists     | [podman-exists(1)](podman-container-exists.1.md)    | Check if a container exists in local storage                                 |
+| exists     | [podman-container-exists(1)](podman-container-exists.1.md)  | Check if a container exists in local storage                         |
```
Comment (rhatdan) [inline docs/podman-container.1.md]: Aphabetize

**Comment by rhatdan** [comment]
@edsantiago Remove refresh and make it hidden.

File: docs/podman-system.1.md (hunk 1 above)
Comment (edsantiago) [reply docs/podman-system.1.md]: There is a `podman system info` command; there is no such man page:
```
$ ll docs/*info*
-rw-rw-r--. 1 esm esm 3933 Feb 28 07:23 docs/podman-info.1
-rw-rw-r--. 1 esm esm 3775 Feb 13 14:18 docs/podman-info.1.md
```
(IOW: this is the `podman-system` man page, and it correctly lists an `info` subcommand, but it is actually documented in `podman-info(1)`. Without my PR, a user would incorrectly believe there is a `podman-system-info(1)` page.)

File: docs/podman.1.md (hunk 2 above)
Comment (edsantiago) [reply docs/podman.1.md]: This one bothers me. Should I just remove it? I mean, the user is already reading `podman(1)`.

File: docs/podman-container.1.md (hunk 3 above)
Comment (edsantiago) [reply docs/podman-container.1.md]: It already is, no? The useful sorting (the one the user cares about) is on the first column; this does not change that. Or have I misunderstood something?

**Comment by edsantiago** [comment]
New force-push: marks `refresh` hidden, and removes it from `podman(1)` and `podman-container(1)`, but leaves `podman-container-refresh(1)` untouched. Please LMK if you think I should remove the self-referential `podman` link in `podman(1)`.

File: docs/podman-system.1.md (hunk 1 above)
Comment (rhatdan) [reply docs/podman-system.1.md]: There is a man page, 

find . -name podman-system-info\*
./docs/links/podman-system-info.1
cat docs/links/podman-system-info.1 
.so man1/podman-info.1

Comment (rhatdan) [reply docs/podman-system.1.md]: Install  the package and run man page and it will work.

File: docs/podman.1.md (hunk 2 above)
Comment (rhatdan) [reply docs/podman.1.md]: Yes

File: docs/podman-system.1.md (hunk 1 above)
Comment (edsantiago) [reply docs/podman-system.1.md]: Oh.... this changes everything. Thank you.

File: docs/podman-container.1.md (hunk 3 above)
Comment (rhatdan) [reply docs/podman-container.1.md]: Yes sorry.

**Comment by rhatdan** [comment]
Remove the podman-container-refresh man page as well.

**Comment by edsantiago** [comment]
New force push. Removed `docs/podman-container-refresh.1.md` as requested. AFAICT this does not need changes to the Makefile -- the doc list is computed via wildcarding. Same with specfile. I may of course have missed something else.

File: docs/podman-system.1.md [hunk 4]
```diff
@@ -13,7 +13,7 @@ The system command allows you to manage the podman systems
 
 | Command  | Man Page                                            | Description                                                                  |
 | -------  | --------------------------------------------------- | ---------------------------------------------------------------------------- |
-| info     | [podman-system-info(1)](podman-info.1.md)           | Displays Podman related system information.                                  |
+| info     | [podman-system-info(1)](podman-system-info.1.md)    | Displays Podman related system information.                                  |
```
Comment (rhatdan) [inline docs/podman-system.1.md]: Sorry leave the link alone.  We ship a link but this link will not work on git.

**Comment by rhatdan** [comment]
One minor nit the LGTM

**Comment by TomSweeneyRedHat** [comment]
Other than @rhatdan 's latest nit, LGTM.

**Comment by edsantiago** [comment]
@rhatdan if I understand your comment in IRC just now, you're asking me to make this change?
```
 | info     | [podman-system-info(1)](podman-info.1.md) |
```
If so, may I humbly push back on that? My reason for wanting it as it is, is for the consistency check (perl oneliner in commit message) to pass cleanly. Although that script is not a CI hook right now, I'd like to spend time next week translating that into something that could be used as a hook.

If I understand your concern, it's that my current way displays inconsistently in the man page: prune and renumber point to `podman-system-xxx` man pages, but info to `podman-info`. Is that your concern? If so, my argument is that the man page reference should be to the canonical command; which in this case is `podman-info`. There is ample precedent for this convention, see `podman-image(1)` and `podman-container(1)`, both of which overwhelmingly refer to `podman-XXX` (canonical names) instead of `podman-subcommand-xxx`.

That said, I will go with whatever you reply.

**Comment by rhatdan** [comment]
Ok you are right, we treated it that way on the others, although all the man pages exists(at least as links.)
/lgtm

**Pull request merged**